qa/tasks/kclient.py

```diff
@@ -109,7 +109,12 @@ def task(ctx, config):
             remote.run(args=["sudo", "bash", "-c", "echo 'module ceph +p' > /sys/kernel/debug/dynamic_debug/control"])
             remote.run(args=["sudo", "bash", "-c", "echo 'module libceph +p' > /sys/kernel/debug/dynamic_debug/control"])
 
-        kernel_mount.mount(mntopts=client_config.get('mntopts', []))
+        try:
+            kernel_mount.mount(mntopts=client_config.get('mntopts', []))
+        except CommandFailedError:
+            log.debug('Rebooting all nodes...')
+            for id_, remote in clients:
+                misc.reboot(remote)
```

Comment:
This is for https://tracker.ceph.com/issues/53288 (@amathuria, please mention the issue in the commit message)

The idea was to reboot nodes when a job failure is detected, to avoid the issue where bare-metal nodes can't be nuked properly with a stale kernel mount, causing us to effectively leak locked/unusable nodes. While it's possible that's a bad idea entirely, it may just be the case that this isn't the wrong _place_ to put the reboot. Would it be better [here](https://github.com/ceph/ceph/blob/master/qa/tasks/kclient.py#L136)?